lib/osut/utils.rb

```diff
@@ -2894,12 +2898,17 @@ def overlaps?(p1 = nil, p2 = nil, flat = true)
     area  = OpenStudio.getArea(union)
     return false if area.empty?
 
-    area = area.get
-    delta = (area - area1 - area2).abs
-    return false if area  < TOL
-    return false if delta < TOL
+    area  = area.get
+    delta = area1 + area2 - area
 
-    true
+    if area > TOL
+      return false if (area - area1).abs < TOL
+      return false if (area - area2).abs < TOL
+      return false if delta.abs < TOL
+      return true  if delta > TOL
+    end
+
+    false
```

Comment:
Revised logic. Catches unexpected (and unfortunately yet untested) cases where Boost-based _join_ returns either polygon area for 2 side-by-side polygons.